core/src/main/java/org/jruby/RubyDir.java

```diff
@@ -84,6 +84,8 @@ public class RubyDir extends RubyObject implements Closeable {
     private int pos;               // current position in directory
     private boolean isOpen = true;
     private Encoding encoding;
+    private static final ReentrantLock lockChdir = new ReentrantLock();
+    private static RubyThread currentThread;
```

Comment:
We cannot store any Ruby objects into a static field in Java because we can have multiple Ruby runtimes existing at the same time.  This needs to be saved in the individual instance of runtime somewhere.